diff --git a/acinclude.m4 b/acinclude.m4
index e1e272f..993d36a 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -1362,10 +1362,10 @@ int main() {
   if (!dir)
     return 1;
   if (readdir_r(dir, (struct dirent *) entry, &pentry) == 0) {
-    close(dir);
+    closedir(dir);
     return 0;
   }
-  close(dir);
+  closedir(dir);
   return 1;
 }
     ]])],[
diff --git a/aclocal.m4 b/aclocal.m4
index 17a24c4..0b695bc 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -1362,10 +1362,10 @@ int main() {
   if (!dir)
     return 1;
   if (readdir_r(dir, (struct dirent *) entry, &pentry) == 0) {
-    close(dir);
+    closedir(dir);
     return 0;
   }
-  close(dir);
+  closedir(dir);
   return 1;
 }
     ]])],[
